codalab/client/local_bundle_client.py

```diff
@@ -332,7 +342,36 @@ def write_targets(self, targets, string):
         for bundle_uuid, subpath in targets:
             if not re.match('^\w+$', subpath):
                 raise UsageError('Can\'t write to subpath with funny characters: %s' % subpath)
-            self.model.add_bundle_action(bundle_uuid, BundleAction.write(subpath, string))
+            
+            if not self.launch_new_worker_system:
+                self.model.add_bundle_action(bundle_uuid, BundleAction.write(subpath, string))
+            else:
+                worker_message = {
+                    'type': 'write',
+                    'uuid': bundle_uuid,
+                    'subpath': subpath,
+                    'string': string,
+                }
+                action_string = BundleAction.write(subpath, string)
+                self._do_bundle_action(bundle_uuid, worker_message, action_string)
```

Comment:
Why does action_string need to have subpath and string inside it, since worker_message already has everything.

Can we make this just self.do_bundle_action(worker_message)?